### ISSUE TITLE
ccdproc requires numpy v1.11.2 - incompatible with v1.12

### ISSUE THREAD

**Issue by bpholden** [Keck-DataReductionPipelines/MosfireDRP#70]
If I use numpy v1.12, I get the following error early on:

```
2017-02-15 22:45:52,308 -        Flats.        handle_flats - INFO: Flat written to combflat_2d_Y.fits
2017-02-15 22:45:52,310 -        Flats.        handle_flats - INFO: Finding slit edges in combflat_2d_Y.fits
Traceback (most recent call last):
  File "Driver.py", line 32, in <module>
    Flats.handle_flats('Flat.txt', maskname, band, flatops)
  File "/Users/holden/anaconda/envs/astroconda/lib/python2.7/site-packages/MOSFIRE-1.0.dev0-py2.7.egg/MOSFIRE/Flats.py", line 155, in handle_flats
    results = find_and_fit_edges(data, header, bs, options,edgeThreshold=edgeThreshold)
  File "/Users/holden/anaconda/envs/astroconda/lib/python2.7/site-packages/MOSFIRE-1.0.dev0-py2.7.egg/MOSFIRE/Flats.py", line 821, in find_and_fit_edges
    threshold_area = vertical_profile[spatial_centers[target]-3:spatial_centers[target]+3]
  File "/Users/holden/anaconda/envs/astroconda/lib/python2.7/site-packages/numpy/ma/core.py", line 3168, in __getitem__
    dout = self.data[indx]
TypeError: slice indices must be integers or None or have an __index__ method
```

Downgrading to numpy v1.11.2 does not have this problem. This could be because of the following issue:
[https://github.com/rbgirshick/py-faster-rcnn/issues/480](url)

**Comment by lucarizzi** [comment]
Thanks Brad 

The pipeline used to give a warning for this error. I guess they decided to upgrade it to an error. The truth is that we need to find out why the index is not an integer as it should be.

**Comment by nickkonidaris** [comment]
Yeah, indexing with mask arrays issued a deprecation warning for a while. Simplest solution is to wrap the mask in a numpy.where().

**Comment by joshwalawender** [comment]
I think (hope) this was resolved by #78 .  If you want to try the solution with a modern numpy version, you can clone the current master branch of the Mosfire DRP repo.
